pkg/provider/azure_controller_vmss.go

```diff
@@ -172,7 +172,7 @@ func (ss *ScaleSet) DetachDisk(ctx context.Context, nodeName types.NodeName, dis
 		// only log here, next action is to update VM status with original meta data
 		klog.Warningf("detach azure disk on node(%s): disk list(%s) not found", nodeName, diskMap)
 	} else {
-		if strings.EqualFold(ss.Environment.Name, consts.AzureStackCloudName) && !ss.Config.DisableAzureStackCloud {
+		if strings.EqualFold(ss.Config.Cloud, consts.AzureStackCloudName) && !ss.Config.DisableAzureStackCloud {
```

Comment:
@MartinForReal any idea why you use `strings.EqualFold(ss.Environment.Name, consts.AzureStackCloudName)` in the beginning?

Reply:
and use ` if ss.Config.IsStackCloud` is better

Reply:
agree +1, good to know we have the util, updated, thank you!

Reply:
Sorry. I didn't notice the IsStackCloud function during the track2 sdk migration.